CatalogService/Controllers/CategoriesController.cs

```diff
@@ -0,0 +1,138 @@
+        {
+            try
+            {
+                return await _getCategoriesHandler.HandleAsync();
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Error while {nameof(GetAllCategoriesAsync)}");
+                throw;
+            }
+        }
+
+        // GET /api/categories/123
+        [HttpGet(template: "{id}", Name = "Get single category")]
+        public async Task<Category> GetCategoryByIdAsync(int id)
+        {
+            try
+            {
+                return await _getCategoryHandler.HandleAsync(id);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Error while {nameof(GetCategoryByIdAsync)}");
+                throw;
+            }
+        }
+
+        // GET /api/categories/123/products
+        [HttpGet(template: "{id}/products", Name = "Get products by category")]
+        public async Task<IEnumerable<Product>> GetProductsByCategoryIdAsync(int id)
+        {
+            try
+            {
+                return await _getProductsByCategoryHandler.HandleAsync(id);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Error while {nameof(GetProductsByCategoryIdAsync)}");
+                throw;
+            }
+        }
+
+        // POST /api/categories
+        [HttpPost(Name = "Add new category")]
+        public async Task<IActionResult> AddCategory([FromBody] CategoryCommand categoryCommand)
+        {
+            try
+            {
+                int id = await _createCategoryHandler.HandleAsync(categoryCommand);
+                return Created($"api/categories/{id}", categoryCommand);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Error while {nameof(AddCategory)}");
+                throw;
+            }
+        }
+
+        // PUT /api/categories/123
+        [HttpPut("{id}", Name = "Update existing category")]
+        public async Task<IActionResult> UpdateCategory(int id, [FromBody] CategoryCommand categoryCommand)
+        {
+            try
+            {
+                Category? updatedCategory = await _updateCategoryHandler.HandleAsync(id, categoryCommand);
+                if (updatedCategory == null)
+                {
+                    return NotFound();
+                }
+
+                return Ok(updatedCategory);
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Error while {nameof(UpdateCategory)}");
+                throw;
+            }
+        }
+
+        // DELETE /api/categories/123
+        [HttpDelete("{id}", Name = "Delete existing category")]
+        public async Task<IActionResult> DeleteCategory(int id)
+        {
+            try
+            {
+                bool categoryExists = await _deleteCategoryHandler.HandleAsync(id);
+                return categoryExists ? Ok() : NotFound();
```

Comment:
Delete should return no content in general. Not found is an inappropriate 
https://stackoverflow.com/questions/6439416/status-code-when-deleting-a-resource-using-http-delete-for-the-second-time

Reply:
Yes, let it be HTTP 204